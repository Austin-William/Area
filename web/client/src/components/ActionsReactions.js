import React from "react";
import { IoIosClose } from 'react-icons/io';
import axios from "axios";
import { url } from "../global/variables.js"
import "../styles/ActionsReactions.css";
// import axios from "axios";

function ActionsReactions(props) {

    const handleDelete = () => {
        let newActionsReactions = props.currentUserData.actionsreactions;
        newActionsReactions.splice(props.index - 1, 1);
        axios.put(url + '/api/user/' + props.currentUserData.id, {
            "actionsreactions": newActionsReactions
        });
    }

    return (
        <tr className="ActionsReactions-tr">
            <th scope="row">{props.index}</th>
            <td>{props.action}</td>
            <td>{props.actionInput}</td>
            <td>{props.reaction}</td>
            <td>
                <button className="Notifications-close-button" type="button" onClick={handleDelete}>
                    <IoIosClose className="Widget-icon" />
                </button>
            </td>
        </tr>
    );
}

export default ActionsReactions;
