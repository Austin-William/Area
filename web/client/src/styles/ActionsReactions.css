.ActionsReactions-tr {
    padding: 1em;
    transition: all 0.1s ease-in-out;
}

.ActionsReactions-tr:hover {
    background-color: #f5f5f5;
}

.ActionsReactions-tr th {
    text-align: left;
    padding-left: 1em;
}
