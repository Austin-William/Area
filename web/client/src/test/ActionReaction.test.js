import React from "react";
import renderer from "react-test-renderer";
import ActionsReactions from "../components/ActionsReactions";

test("Components : ActionsReactions renders correctly", () => {
  const tree = renderer.create(<ActionsReactions />).toJSON();
  expect(tree).toMatchSnapshot();
});

