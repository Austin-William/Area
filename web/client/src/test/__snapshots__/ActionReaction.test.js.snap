// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : ActionsReactions renders correctly 1`] = `
<div
  className="row"
>
  Action = undefined Reaction = undefined
</div>
`;
